test_conformance/extensions/cl_khr_command_buffer/command_buffer_printf.cpp

```diff
@@ -0,0 +1,556 @@
+    {
+        cl_int error = CL_SUCCESS;
+
+        error = clCommandNDRangeKernelKHR(
+            command_buffer, nullptr, nullptr, kernel, 1, nullptr, &num_elements,
+            nullptr, 0, nullptr, nullptr, nullptr);
+        test_error(error, "clCommandNDRangeKernelKHR failed");
+
+        error = clFinalizeCommandBufferKHR(command_buffer);
+        test_error(error, "clFinalizeCommandBufferKHR failed");
+        return CL_SUCCESS;
+    }
+
+    //--------------------------------------------------------------------------
+    int WaitForEvent(cl_event* event)
+    {
+        cl_int status = clWaitForEvents(1, event);
+        if (status != CL_SUCCESS)
+        {
+            log_error("clWaitForEvents failed");
+            return status;
+        }
+
+        status = clReleaseEvent(*event);
+        if (status != CL_SUCCESS)
+        {
+            log_error("clReleaseEvent failed. (*event)");
+            return status;
+        }
+        return CL_SUCCESS;
+    }
+
+    //--------------------------------------------------------------------------
+#define test_error_release_stdout(errCode, msg)                                \
+    {                                                                          \
+        auto errCodeResult = errCode;                                          \
+        if (errCodeResult != CL_SUCCESS)                                       \
+        {                                                                      \
+            ReleaseOutputStream(file_descriptor);                              \
+            print_error(errCodeResult, msg);                                   \
+            return errCode;                                                    \
+        }                                                                      \
+    }
+
+    //--------------------------------------------------------------------------
+    cl_int EnqueueSinglePass(const std::vector<cl_char>& pattern,
+                             std::vector<cl_char>& output_data)
+    {
+        cl_int error = CL_SUCCESS;
+        auto in_mem_size = sizeof(cl_char) * pattern.size();
+        error = clEnqueueWriteBuffer(queue, in_mem, CL_TRUE, 0, in_mem_size,
+                                     &pattern[0], 0, nullptr, nullptr);
+        test_error(error, "clEnqueueFillBuffer failed");
+
+        cl_int offset[] = { 0, pattern.size() - 1 };
+        error = clEnqueueWriteBuffer(queue, off_mem, CL_TRUE, 0, sizeof(offset),
+                                     offset, 0, nullptr, nullptr);
+        test_error(error, "clEnqueueFillBuffer failed");
+
+        // redirect output stream to temporary file
+        file_descriptor = AcquireOutputStream(&error);
+        if (error != 0)
+        {
+            log_error("Error while redirection stdout to file");
+            return TEST_FAIL;
+        }
+
+        // enqueue command buffer with kernel containing printf command
+        cl_event wait_event;
```

Comment:
Using `clEventWrapper` rather than `cl_event` should release `wait_event` when it goes out of scope